src/packtPublishingFreeEbook.py

```diff
@@ -128,6 +128,14 @@ def __create_http_session(self):
             raise requests.exceptions.RequestException(message)
         logger.info("Session created, logged in successfully!")
 
+    def __exists_book(self, title):
+        # TODO: reduce fetch pages
+        my_books_data = self.get_all_books_data()
+        for data in my_books_data:
+            if ConfigurationModel.convert_book_title_to_valid_string(data['title']) == \
+                    ConfigurationModel.convert_book_title_to_valid_string(title):
+                return True
```

Comment:
It's not very pythonic, it reminds me of how it would be done in old school C or C++. We could do it using the iterator.

```
valid_book_title = ConfigurationModel.convert_book_title_to_valid_string
return any(valid_book_title(title) == valid_book_title(data['title']) for data in self.get_all_books_data())
```

However, if you prefer to do it old school way then I won't insist.